packages/splits-sdk/src/client/warehouse.ts

```diff
@@ -0,0 +1,1487 @@
+      ensPublicClient,
+    })
+
+    this.estimateGas = new WarehouseGasEstimates({
+      chainId,
+      publicClient,
+      walletClient,
+      includeEnsNames,
+      ensPublicClient,
+    })
+
+    this.sign = new WarehouseSignature({
+      chainId,
+      publicClient,
+      walletClient,
+      includeEnsNames,
+      ensPublicClient,
+    })
+  }
+
+  // ERC6909 Metadata
+  async getName({ tokenAddress }: { tokenAddress: Address }): Promise<{
+    name: string
+  }> {
+    validateAddress(tokenAddress)
+
+    this._requirePublicClient()
+
+    const name = await this._warehouseContract.read.name([
+      fromHex(tokenAddress, 'bigint'),
+    ])
+
+    return {
+      name,
+    }
+  }
+
+  async getSymbol({ tokenAddress }: { tokenAddress: Address }): Promise<{
+    symbol: string
+  }> {
+    validateAddress(tokenAddress)
+
+    this._requirePublicClient()
+
+    const symbol = await this._warehouseContract.read.symbol([
+      fromHex(tokenAddress, 'bigint'),
+    ])
+
+    return {
+      symbol,
+    }
+  }
+
+  async getDecimals({ tokenAddress }: { tokenAddress: Address }): Promise<{
+    decimals: number
+  }> {
+    validateAddress(tokenAddress)
+
+    this._requirePublicClient()
+
+    const decimals = await this._warehouseContract.read.decimals([
+      fromHex(tokenAddress, 'bigint'),
+    ])
+
+    return {
+      decimals,
+    }
+  }
+
+  // Warehouse withdraw config
+  async getWithdrawConfig({
+    userAddress,
+  }: {
+    userAddress: Address
+  }): Promise<{ withdrawConfig: { incentive: number; paused: boolean } }> {
+    validateAddress(userAddress)
+
+    this._requirePublicClient()
+
+    const config = await this._warehouseContract.read.withdrawConfig([
+      userAddress,
+    ])
+
+    return {
+      withdrawConfig: {
+        incentive: config[0],
+        paused: config[1],
+      },
+    }
+  }
+
+  // ERC6909X nonce
+  async isValidNonce({
+    userAddress,
+    userNonce,
+  }: {
+    userAddress: Address
+    userNonce: bigint
+  }): Promise<{ isValidNonce: boolean }> {
+    validateAddress(userAddress)
+
+    this._requirePublicClient()
+
+    const isValidNonce = await this._warehouseContract.read.isValidNonce([
+      userAddress,
+      userNonce,
+    ])
+
+    return {
+      isValidNonce,
+    }
+  }
+
+  async eip712Domain(): Promise<{ domain: TypedDataDomain }> {
+    return this._eip712Domain()
+  }
+
+  // ERC6909 read
+  async isOperator({
+    ownerAddress,
+    operatorAddress,
+  }: {
+    ownerAddress: Address
+    operatorAddress: Address
+  }): Promise<{ isOperator: boolean }> {
+    validateAddress(ownerAddress)
+    validateAddress(operatorAddress)
+
+    this._requirePublicClient()
+
+    const isOperator = await this._warehouseContract.read.isOperator([
+      ownerAddress,
+      operatorAddress,
+    ])
+
+    return {
+      isOperator,
+    }
+  }
+
+  async balanceOf({
+    ownerAddress,
+    tokenAddress,
+  }: {
+    ownerAddress: Address
+    tokenAddress: Address
```

Comment:
nit: i feel like we're kinda inconsistent with our naming (this probably exists in the current sdk, not a new thing).

Feel like this is a good opportunity to get consistent. What do you think? Basically i'm taking about stuff like `token` vs `tokenAddress`. Feel like we should use the same pattern everywhere. Maybe we just drop the `Address` part? what do you think? Is there one you like better?

Reply:
I think we should use xAddress everywhere ( I am guilty of not doing this). Token might become an object in the future same with owner, split, creator.